$OpenBSD: patch-mcs_class_System_System_Platform_cs,v 1.1 2019/03/25 13:05:39 thfr Exp $

add bool for Platform.IsOpenBSD, to address
https://github.com/mono/mono/issues/8168
Upstreamed in PR: https://github.com/mono/mono/pull/13633

Index: mcs/class/System/System/Platform.cs
--- mcs/class/System/System/Platform.cs.orig
+++ mcs/class/System/System/Platform.cs
@@ -48,6 +48,7 @@ namespace System {
 
 #else
 		static bool isFreeBSD;
+		static bool isOpenBSD;
 
 		[DllImport ("libc")]
 		static extern int uname (IntPtr buf);
@@ -68,6 +69,9 @@ namespace System {
 				case "FreeBSD":
 					isFreeBSD = true;
 					break;
+				case "OpenBSD":
+					isOpenBSD = true;
+					break;
 				}
 			}
 			Marshal.FreeHGlobal (buf);
@@ -88,6 +92,14 @@ namespace System {
 				if (!checkedOS)
 					CheckOS();
 				return isFreeBSD;
+			}
+		}
+
+		public static bool IsOpenBSD {
+			get {
+				if (!checkedOS)
+					CheckOS();
+				return isOpenBSD;
 			}
 		}
 	}
